src/stores/VoiceChannelStore.ts

```diff
@@ -0,0 +1,246 @@
+        this.assertConnected();
+
+        const waitForMute = new Promise<void>(resolve =>
+            this.activeChannel.once(`action:${ElementWidgetActions.MuteVideo}`, () => resolve()),
+        );
+        this.activeChannel.transport.send(ElementWidgetActions.MuteVideo, {});
+        await this.timeout(waitForMute);
+    };
+
+    public unmuteVideo = async () => {
+        this.assertConnected();
+
+        const waitForUnmute = new Promise<void>(resolve =>
+            this.activeChannel.once(`action:${ElementWidgetActions.UnmuteVideo}`, () => resolve()),
+        );
+        this.activeChannel.transport.send(ElementWidgetActions.UnmuteVideo, {});
+        await this.timeout(waitForUnmute);
+    };
+
+    private assertConnected = () => {
+        if (!this.activeChannel) throw new Error("Not connected to any voice channel");
+    };
+
+    private timeout = async (wait: Promise<void>) => {
+        if (await timeout(wait, false, VoiceChannelStore.TIMEOUT) === false) {
+            throw new Error("Communication with voice channel timed out");
+        }
+    };
+
+    private ack = (ev: CustomEvent<IWidgetApiRequest>) => {
+        this.activeChannel.transport.reply(ev.detail, {});
+    };
+
+    private onHangup = (ev: CustomEvent<IWidgetApiRequest>) => {
+        this.ack(ev);
+
+        this.activeChannel.off(`action:${ElementWidgetActions.CallParticipants}`, this.onParticipants);
+        this.activeChannel.off(`action:${ElementWidgetActions.MuteAudio}`, this.onMuteAudio);
+        this.activeChannel.off(`action:${ElementWidgetActions.UnmuteAudio}`, this.onUnmuteAudio);
+        this.activeChannel.off(`action:${ElementWidgetActions.MuteVideo}`, this.onMuteVideo);
+        this.activeChannel.off(`action:${ElementWidgetActions.UnmuteVideo}`, this.onUnmuteVideo);
+
+        this.activeChannel = null;
+        this._roomId = null;
+        this._participants = null;
+        this._audioMuted = null;
+        this._videoMuted = null;
+
+        this.emit(VoiceChannelEvent.Disconnect);
+    };
+
+    private onParticipants = (ev: CustomEvent<IWidgetApiRequest>) => {
+        this.ack(ev);
+
+        this._participants = ev.detail.data.participants as IJitsiParticipant[];
+        this.emit(VoiceChannelEvent.Participants, ev.detail.data.participants);
+    };
+
+    private onMuteAudio = (ev: CustomEvent<IWidgetApiRequest>) => {
+        this.ack(ev);
+
+        this._audioMuted = true;
+        this.emit(VoiceChannelEvent.MuteAudio);
+    };
+
+    private onUnmuteAudio = (ev: CustomEvent<IWidgetApiRequest>) => {
+        this.ack(ev);
+
+        this._audioMuted = false;
+        this.emit(VoiceChannelEvent.UnmuteAudio);
+    };
+
+    private onMuteVideo = (ev: CustomEvent<IWidgetApiRequest>) => {
+        this.ack(ev);
+
+        this._videoMuted = true;
+        this.emit(VoiceChannelEvent.MuteVideo);
+    };
+
+    private onUnmuteVideo = (ev: CustomEvent<IWidgetApiRequest>) => {
+        this.ack(ev);
```

Comment:
like the element-web side: the acks should be the last thing in these handlers. While the handlers are async, it means that the widget will get a timeout error which isn't important for how we're using the API specifically, but is helpful in logs and such.